builtin/providers/azurerm/resource_arm_storage_account.go

```diff
@@ -131,16 +132,47 @@ func resourceArmStorageAccountCreate(d *schema.ResourceData, meta interface{}) e
 		Tags:     expandTags(tags),
 	}
 
-	_, err := storageClient.Create(resourceGroupName, storageAccountName, opts, make(chan struct{}))
-	if err != nil {
-		return fmt.Errorf("Error creating Azure Storage Account '%s': %s", storageAccountName, err)
+	// Create the storage account. We wrap this so that it is cancellable
+	// with a Ctrl-C since this can take a LONG time.
+	wrap := signalwrapper.Run(func(cancelCh <-chan struct{}) error {
+		_, err := storageClient.Create(resourceGroupName, storageAccountName, opts, cancelCh)
+		return err
+	})
+
+	// Check the result of the wrapped function. I put this into a select
+	// since we will likely also want to introduce a time-based timeout.
+	var createErr error
+	select {
+	case <-time.After(1 * time.Hour):
+		// An hour is way above the expected P99 for this API call so
+		// we premature cancel and error here.
+		createErr = wrap.Cancel()
+	case createErr = <-wrap.ErrCh:
+		// Successfully ran (but perhaps not successfully completed)
+		// the function.
 	}
 
 	// The only way to get the ID back apparently is to read the resource again
 	read, err := storageClient.GetProperties(resourceGroupName, storageAccountName)
 	if err != nil {
 		return err
 	}
+
+	// Set the ID right away if we have one
+	if read.ID != nil {
+		log.Printf("[INFO] storage account %q ID: %q", storageAccountName, *read.ID)
+		d.SetId(*read.ID)
+	}
+
+	// If we had a create error earlier then we return with that error now.
+	// We do this later here so that we can grab the ID above is possible.
+	if createErr != nil {
+		return fmt.Errorf(
+			"Error creating Azure Storage Account '%s': %s",
+			storageAccountName, createErr)
+	}
```

Comment:
Should this check be before the follow-on read? Thinking that if the read 404s we mask the create error...